packages/lesswrong/components/recentDiscussion/useRecentDiscussionThread.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback, useState } from "react";
+import { useRecordPostView } from "../hooks/useRecordPostView";
+import { ThreadableCommentType, unflattenComments } from "../../lib/utils/unflatten";
+import type { CommentTreeOptions } from "../comments/commentTree";
+
+export const useRecentDiscussionThread = <T extends ThreadableCommentType>({
+  post,
+  comments,
+  refetch,
+  commentTreeOptions = {},
+  initialExpandAllThreads,
+} : {
+  post: PostsRecentDiscussion,
+  comments?: T[],
+  refetch: () => void,
+  commentTreeOptions?: CommentTreeOptions,
+  initialExpandAllThreads?: boolean,
+}) => {
+  const [highlightVisible, setHighlightVisible] = useState(false);
+  const [markedAsVisitedAt, setMarkedAsVisitedAt] = useState<Date|null>(null);
+  const [expandAllThreads, setExpandAllThreads] = useState(initialExpandAllThreads ?? false);
+  const {recordPostView} = useRecordPostView(post);
+
+  const markAsRead = useCallback(
+    () => {
+      setMarkedAsVisitedAt(new Date());
+      setExpandAllThreads(true);
+      recordPostView({post, extraEventProperties: {type: "recentDiscussionClick"}})
+    },
+    [setMarkedAsVisitedAt, setExpandAllThreads, recordPostView, post],
+  );
+  const showHighlight = useCallback(
+    () => {
+      setHighlightVisible(!highlightVisible);
+      markAsRead();
+    },
+    [setHighlightVisible, highlightVisible, markAsRead],
+  );
+
+  const lastCommentId = comments && comments[0]?._id;
+  const nestedComments = unflattenComments<T>(comments ?? []);
+
+  const lastVisitedAt = markedAsVisitedAt || post.lastVisitedAt
+
+  // TODO verify whether/how this should be interacting with afCommentCount
```

Comment:
This comment is actually just moved from when this was refactored out of `RecentDiscussionThread`. I think there's a good chance it's out of date, but since it doesn't actually pertain to us I didn't want to remove it.